src/components/chat/tabs/account-tab/add-entry-form.tsx

```
@@ -0,0 +1,216 @@
+  })
+
+export class AddEntryForm extends React.Component<
+  { memberId: string; showNotification: (data: any) => void },
+  State
+> {
+  public state = {
+    confirmed: false,
+  }
+
+  public render() {
+    return (
+      <Mutation
+        mutation={ADD_ACCOUNT_ENTRY_MUTATION}
+        refetchQueries={() => [
+          {
+            query: GET_MEMBER_ACCOUNT_QUERY,
+            variables: { memberId: this.props.memberId },
+          },
+        ]}
+      >
+        {(mutation, { loading }) => (
+          <Formik
+            initialValues={{
+              type: '',
+              amount: '',
+              reference: '',
+              source: '',
+              title: '',
+              comment: '',
+              fromDate: new Date(),
+            }}
+            onSubmit={(formData: any, { resetForm }) => {
+              if (!this.state.confirmed || loading) {
+                return
+              }
+              mutation({
+                variables: {
+                  memberId: this.props.memberId,
+                  accountEntry: {
+                    type: formData.type,
+                    amount: {
+                      amount: parseFloat(formData.amount),
+                      currency: 'SEK',
+                    },
+                    fromDate: format(formData.fromDate, 'yyyy-MM-dd'),
+                    reference: formData.reference,
+                    source: formData.source,
+                    title: formData.title,
+                    comment: formData.comment,
+                  },
+                },
+              })
+                .then(() => {
+                  this.props.showNotification({
+                    message: 'Account entry added.',
+                    header: 'Success',
+                    type: 'olive',
+                  })
+                  resetForm()
+                })
+                .catch((error) => {
+                  this.props.showNotification({
```

maybe we should throw again? just so we don't swallow the error